scripts/release/list_backports.py

```diff
@@ -0,0 +1,340 @@
+                        help='print output in JSON rather than RST')
+    parser.add_argument('-s', '--start', dest='start', help='start date (YYYY-mm-dd)',
+                        metavar='START_DATE', type=valid_date_type)
+    parser.add_argument('-e', '--end', dest='end', help='end date (YYYY-mm-dd)',
+                        metavar='END_DATE', type=valid_date_type)
+    parser.add_argument("-o", "--org", default="zephyrproject-rtos",
+                        help="Github organisation")
+    parser.add_argument('-p', '--include-pull', dest='includes',
+                        help='include pull request (can be specified multiple times)',
+                        metavar='PR', type=int, action='append', default=[])
+    parser.add_argument('-P', '--exclude-pull', dest='excludes',
+                        help='exlude pull request (can be specified multiple times, helpful for version bumps and release notes)',
+                        metavar='PR', type=int, action='append', default=[])
+    parser.add_argument("-r", "--repo", default="zephyr",
+                        help="Github repository")
+
+    args = parser.parse_args()
+
+    if args.includes:
+        if getattr(args, 'start'):
+            logging.error(
+                'the --start argument should not be used with --include-pull')
+            return None
+        if getattr(args, 'end'):
+            logging.error(
+                'the --end argument should not be used with --include-pull')
+            return None
+    else:
+        if not getattr(args, 'start'):
+            logging.error(
+                'if --include-pr PR is not used, --start START_DATE is required')
+            return None
+
+        if not getattr(args, 'end'):
+            setattr(args, 'end', datetime.now())
+
+        if args.end < args.start:
+            logging.error(
+                f'end date {args.end} is before start date {args.start}')
+            return None
+
+    if args.tokenfile:
+        with open(args.tokenfile, 'r') as file:
+            token = file.read()
+            token = token.strip()
+    else:
+        if 'GITHUB_TOKEN' not in os.environ:
+            raise ValueError('No credentials specified')
+        token = os.environ['GITHUB_TOKEN']
+
+    setattr(args, 'token', token)
+
+    return args
+
+
+class Backport(object):
+    def __init__(self, repo, base, pulls):
+        self._base = base
+        self._repo = repo
+        self._issues = []
+        self._pulls = pulls
+
+        self._pulls_without_an_issue = []
+        self._pulls_with_invalid_issues = {}
+
+    @staticmethod
+    def by_date_range(repo, base, start_date, end_date, excludes):
+        """Create a Backport object with the provided repo,
+        base, start datetime object, and end datetime objects, and
+        list of excluded PRs"""
+
+        pulls = []
+
+        unfiltered_pulls = repo.get_pulls(
+            base=base, state='closed')
+        for p in unfiltered_pulls:
```

Comment:
These takes forever for me, are the APIs just that slow? Wondering if it's worth adding a verbose flag to print `unfiltered_pulls.totalCount` here and then a line as PR get fetched, what do you think? (can't see any way of making it faster but maybe I'm missing something)

Reply:
Yeah it does not look like it's possible using that API right now, though `list_issues.py` does that using `requests.get` directly for issues: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/release/list_issues.py#L109, worth looking into it? Probably not worth it for something we run once per release.

Reply:
It might be worth optimizing that at some point later, or maybe requesting GH add that feature to their API, but yeah, probably not pressing right now.